benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -1593,9 +1607,9 @@ void GraphicsBenchmarkApp::RecordCommandBufferSpheres(PerFrame& frame)
     frame.cmd->BindGraphicsDescriptorSets(mSphere.pipelineInterface, 1, &mSphere.descriptorSets.at(GetInFlightFrameIndex()));
 
     // Snapshot some scene-related values for the current frame
-    uint32_t currentSphereCount   = pSphereInstanceCount->GetValue();
+    uint32_t currentSphereCount   = std::min<uint32_t>(pSphereInstanceCount->GetValue(), mInitializedSpheres);
     uint32_t currentDrawCallCount = pDrawCallCount->GetValue();
-    uint32_t mSphereIndexCount    = mSphereMeshes[meshIndex]->GetIndexCount() / kMaxSphereInstanceCount;
+    uint32_t mSphereIndexCount    = (mInitializedSpheres > 0 ? (mSphereMeshes[meshIndex]->GetIndexCount() / mInitializedSpheres) : 0);
```

Comment:
in what case `mInitializedSpheres` could be 0? If i understand correctly, only when `SetupSphereMeshes` is skipped,  mInitializedSpheres is 0? but in that case, `pEnableSpheres` is false, and  `RecordCommandBufferSpheres` would be skipped as well?

Reply:
I don't think those case could be reached as-is. But since those values depends on elements external to this function, I would be in favor of keeping this check and min.
Just so we say "in case this happens, we have a defined behavior to handle it", even if if wouldn't happen today.

Reply:
if `mInitializedSpheres` should never be 0, maybe better with an assert?

Reply:
Updated.

Reply:
Yes, an assert is also fine if that's an illegal behavior.
But I'd feel like having the slider default to 0 is a valid behavior.

Reply:
the min value of pSphereInstanceCount is set to 1, so i guess 0 is not valid.

Reply:
actually, maybe it is better to skip `RecordCommandBufferSpheres()` entirely if `mInitializedSpheres` is 0?

Reply:
I agree with that change. Updated.